app/mailers/concerns/deliverable.rb

```diff
@@ -0,0 +1,16 @@
+module Deliverable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_perform_deliveries
+    after_action  :set_delivery_options
+  end
+
+  def set_perform_deliveries
+    self.perform_deliveries = ForemInstance.smtp_enabled?
+  end
+
+  def set_delivery_options
+    mail.delivery_method.settings.merge!(Settings::SMTP.settings)
```

Comment:
How does this incorporate `SENDGRID_API_KEY`?

Reply:
`Settings::SMTP.settings` returns a hardcoded config if `SENDGRID_API_KEY` exists.

See https://github.com/forem/forem/blob/main/app/models/settings/smtp.rb#L20